monitoring/grafana/dashboards/test-jsonnet.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+set -e
+TEMPDIR=`mktemp -d`
+BASEDIR=$(dirname "$0")
+
+jsonnet -m ${TEMPDIR} $BASEDIR/jsonnet/grafana_dashboards.jsonnet
```

Comment:
I've simplified the script and added a feature which will print a traceback and exit with 1 if any of the JSON files invalid. That output was suppressed due to `set -e` (as I suggested). You may use (and adapt) it, if you want.

```bash
#!/usr/bin/env bash

# Run this script in src/pybind/mgr/dashboard with `tox -e grafonnet` if you
# have any issues running this script directly.

set -e

tmpdir=$(mktemp -d)
clean() {
    rm ${tmpdir} -rf
}
trap clean 0
trap clean 1

cd $(dirname "$0")

for gen_json in $(jsonnet -m ${tmpdir} jsonnet/grafana_dashboards.jsonnet); do
    base_name=$(basename $gen_json)
    # jsondiff exits with exit code 1 if both, there's a difference between two
    # files AND if it throws a Traceback due to the JSON of one file isn't
    # valid, so we check the output instead of the exit code.
    diff=$(jsondiff --indent 2 $gen_json $base_name 2>&1 || true)

    if [[ "$diff" =~ ^Traceback ]]; then
        echo -e "$diff"
        exit 1
    elif [[ "$diff" != "{}" ]] ; then
        echo "failed comparing generated file with existing, please re-generate $base_name"
        echo "$diff"
        exit 1
    fi
done

echo "success"
```

Reply:
This script works better, Thanks for this @p-se .